### ISSUE TITLE
Customer Addresses: Can't convert undefined to object

### ISSUE THREAD

**Issue by leoquijano** [magento/magento2#7115]
Hello,

I'm using Magento 2.1.2, and I noticed that there's a Javascript error when loading the `vendor/magento/module-customer/view/frontend/web/js/model/customer-addresses.js` file, in the product detail page.

I get this: 

![captura de pantalla de 2016-10-20 20-19-01](https://cloud.githubusercontent.com/assets/7789514/19584554/985ee010-9702-11e6-9a1d-58c8d8ad5e62.png)

After reviewing the source, I noticed that this is the failing code:

``` javascript
define(
    [
        'jquery',
        'ko',
        './customer/address'
    ],
    function($, ko, address) {
        "use strict";
        var isLoggedIn = ko.observable(window.isCustomerLoggedIn);
        return {
            getAddressItems: function() {
                var items = [];
                if (isLoggedIn) {
                    var customerData = window.customerData;
                    if (Object.keys(customerData).length) {
                        $.each(customerData.addresses, function (key, item) {
                            items.push(new address(item));
                        });
                    }
                }
                return items;
            }
        }
    }
);
```

Note how isLoggedIn is called without parentheses, even though it is a Knockout observable ([http://knockoutjs.com/documentation/observables.html](documentation here)):

``` javascript
if (isLoggedIn) {
```

Since the existence of `window.customerData` relies on whether or not the customer is logged in, it's important to get this call right.

After replacing that single line in `customer-addresses.js` with this:

``` javascript
if (isLoggedIn()) {
```

... the problem goes away.

**Comment by odubovyk** [comment]
Hi @leoquijano, thank you for the report.
Could you please add the Steps To Reproduce of how you come to error in console.

**Comment by leoquijano** [comment]
Hi @odubovyk, open the product detail page while logged out. It may also break in the checkout page.

If it's not happening due to other conditions (i'm using a custom theme), try to call that function from a template, `getAddressItems`, while not logged in. 

You can see directly in the code example that the observable is not being used correctly, anyway. Calling `isLoggedIn` without the parentheses, will return you the observable function, and not the observed value.

**Comment by odubovyk** [comment]
Thank you. Internal ticket - MAGETWO-60128

**Comment by korostii** [comment]
Hi @magento-team,
Are there any plans to backport [the fix](https://github.com/magento/magento2/commit/71f4018a127714a185584176675a654ad6632866#commitcomment-22685718) to 2.1?

**Comment by magento-engcom-team** [comment]
@leoquijano, thank you for your report.
The issue is already fixed in 2.2.0
